### PR TITLE
add soft asserts when calling put/fetch/delete_attachments

### PR DESCRIPTION
**Pull request by nickpell**
This is a bit paranoid, but I'd like to be confident that if I were to remove the ```*_attachments``` from couchdbkit, we wouldn't get a flood of 500s.  It's less helpful in preventing breaking some rarely used and untested code, but that would be a less disastrous failure mode.

### CODE REVIEW

**Review by emord — APPROVED** [review]
+1 for paranoia

File: corehq/ex-submodules/dimagi/ext/couchdbkit.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
     DateTimeProperty, OldDateTimeProperty
 from dimagi.utils.couch.database import SafeSaveDocument
```
[the remark this answers is not in the document]
Comment (nickpell) [reply corehq/ex-submodules/dimagi/ext/couchdbkit.py]: Agreed

**Review by pr33thi — APPROVED** [review]

**Review by millerdev — APPROVED** [review]
Would it make sense to also soft assert on `save` if the `_attachments` key in the document is not null or empty?

**Comment by nickpell** [comment]
Based on tests, we should get ```CommCareBuild``` attachments into blob storage.

**Comment by nickpell** [comment]
> Would it make sense to also soft assert on save if the _attachments key in the document is null or empty.

For purposes of getting off of restkit it's actually ok if there are couch attachments, as long as none of these 3 functions are used to access them.  If the ```_attachments``` key is used, that will still be supported by couchdbkit.  While it would be good to eventually get all attachments out of couch, I'm treating migrating couch attachments not accessed via these 3 functions as out-of-scope.

**Pull request merged**